packages/node-terra/src/indexer/apiterra.service.ts

```diff
@@ -76,10 +89,54 @@ export class TerraClient {
     return this.baseApi.tendermint.blockInfo(height, this.params);
   }
 
+  // eslint-disable-next-line @typescript-eslint/require-await
+  async blockInfoMantlemint(height?: number): Promise<BlockInfo> {
+    const config = {
+      method: 'get',
+      url: `${this.mantlemintURL}/index/blocks/${height}`,
+    };
+    return axios(config)
+      .then((d) => d.data)
+      .catch((e) => {
+        throw e;
+      });
+  }
+
   async txInfo(hash: string): Promise<TxInfo> {
     return this.baseApi.tx.txInfo(hashToHex(hash), this.params);
   }
 
+  // eslint-disable-next-line @typescript-eslint/require-await
+  async mantlemintHealthCheck(): Promise<boolean> {
+    logger.info(this.mantlemintURL);
+    if (!this.mantlemintURL) {
+      return false;
+    }
+    const config = {
+      method: 'get',
+      url: `${this.mantlemintURL}/health`,
+    };
+    const health = axios(config)
+      .then((d) => d.data)
+      .catch((e) => {
+        throw e;
+      });
+    return health === 'OK';
+  }
+
+  // eslint-disable-next-line @typescript-eslint/require-await
+  async txsByHeightMantlemint(height: string): Promise<TxInfo[]> {
```

Comment:
How sure are we on this return type? Could there be differences in encoding? Or anything that needs to be parsed?

Reply:
The returned object had the same set of keys as in TxInfo